api/src/utils/apply-query.ts

```diff
@@ -221,7 +221,7 @@ export function applyFilter(
 							.on(
 								`${parentAlias || parentCollection}.${relation.field}`,
 								'=',
-								knex.raw(`CAST(?? AS VARCHAR(255))`, `${alias}.${schema.collections[pathScope].primary}`)
+								knex.raw(`CAST(?? AS CHAR)`, `${alias}.${schema.collections[pathScope].primary}`)
```

Comment:
cc @Oreilles

Reply:
`CHAR` works for all vendors except Postgres, which returns only one character. Casting to `CHAR(255)` instead seems to make everyone happy.